docs/src/architecture/accounts.md

```diff
@@ -5,40 +5,52 @@ Miden aims to support expressive smart contracts via a Turing-complete language.
-* **Storage &rarr;** a user-defined data which can be stored in an account
-* **Nonce &rarr;** a counter which must be incremented whenever account state changes
+* **Storage &rarr;** user-defined data which can be stored in an account
+* **Nonce &rarr;** a counter which must be incremented whenever the account state changes
 * **Vault &rarr;** a collection of assets stored in an account
-* **Code &rarr;** a collection of functions which define an external interface for an account
+* **Code &rarr;** a collection of functions which define the external interface for an account
 
 ### Account ID
-~63 bits (1 Felt) long identifier for the account. The first three significant bits specify its type and the [storage mode](https://0xpolygonmiden.github.io/miden-base/architecture/accounts.html#account-storage-modes). There are four types of accounts in Miden:
+~63 bits (1 Felt) long identifier for the account. The four most significant bits specify its type and the [storage mode](https://0xpolygonmiden.github.io/miden-base/architecture/accounts.html#account-storage-modes). 
+
+#### Account types
+There are four types of accounts in Miden:
 
 | | Regular updatable account | Regular immutable account | Faucet for fungible assets | Faucet for non-fungible assets |
 |---|---|---|---|---|
-| **Description** | Will be used by most users for a wallet. Code specification and changes possible. | Will be used by most smart contracts. Once deployed code cannot be changed | Users can issue fungible assets and customize them. | Users can issue non-fungible assets and customize them. |
+| **Description** | For most users, e.g. a wallet. Code specification and change possible. | For most smart contracts. Once deployed code is immutable. | Users can issue fungible assets and customize them. | Users can issue non-fungible assets and customize them. |
 | **Code updatability** | yes | no | no | no |
 | **Most significant bits** | `00` | `01` | `10` | `11` |
 
-#### Example account ID (big-endian)
+#### Account storage mode
+Account data - stored by the Miden Node - can be public, private, or encrypted. The third and fourth most significant bits of the account ID specifies whether the account data is public `00`, encrypted `01`, or private `11`.
+
+* Accounts with **public state**, where the actual state is stored onchain. These would be similar to how accounts work in public blockchains. Smart contracts that depend on public shared state should be stored public on Miden, e.g., DEX contract.
+* Account with **encrypted state**, where the account data is stored onchain but in encrypted text. It provides liveness guarantee of the protocol for the account in question.
+* Accounts with **private state**, where only the hash of the account is stored onchain. Users who want stay private and take care of their own data should choose this mode. The hash is defined as: `hash([account ID, 0, 0, nonce], [vault root], [storage root], [code root])`.
```

Comment:
I would probably separate encrypted state from the other two for now as we don't yet support it (and we still need to figure out how exactly we want to do this). Maybe say something like: "In the future, we'll also support ...".